python/dgl/graph_serialize.py

```diff
@@ -0,0 +1,63 @@
+"""For Graph Serialization"""
+from __future__ import absolute_import
+from .graph import DGLGraph
+from ._ffi.object import ObjectBase, register_object
+# from ._ffi.base import c_array
+from . import _api_internal
+from ._ffi.function import _init_api
+from . import backend as F
+
+_init_api("dgl.graph_serialize")
+
+
+@register_object("graph_serialize.GraphData")
+class GraphData(ObjectBase):
+    """GraphData Object"""
+
+    @staticmethod
+    def create(g: DGLGraph):
+        """Create GraphData"""
+        ghandle = g._graph
+        node_tensors = dict()
+        edge_tensors = dict()
+        for key, value in g.ndata.items():
+            node_tensors[key] = F.zerocopy_to_dgl_ndarray(value)
+        for key, value in g.edata.items():
+            edge_tensors[key] = F.zerocopy_to_dgl_ndarray(value)
+        return _CAPI_MakeGraphData(ghandle, node_tensors, edge_tensors)
+
+    def get_graph(self):
+        """Get DGLGraph from GraphData"""
+        ghandle = _CAPI_GDataGraphHandle(self)
+        g = DGLGraph(graph_data=ghandle, readonly=True)
+        node_tensors_items = _CAPI_GDataNodeTensors(self).items()
+        edge_tensors_items = _CAPI_GDataEdgeTensors(self).items()
+        for k, v in node_tensors_items:
+            g.ndata[k] = F.zerocopy_from_dgl_ndarray(_api_internal._ValueGet(v))
+        for k, v in edge_tensors_items:
+            g.edata[k] = F.zerocopy_from_dgl_ndarray(_api_internal._ValueGet(v))
+        return g
+
+
+def save_graphs(filename, g_list):
+    """
+    Save DGLGraphs to file
+    :param filename: file to store DGLGraphs
+    :param g_list: list of DGLGraph
+    :return:
+    """
+    gdata_list = [GraphData.create(g) for g in g_list]
+    _CAPI_DGLSaveGraphs(filename, gdata_list)
```

Comment:
One question. The Object FFI is not very efficient for large list. Do you expect the given list to be large? We might need some benchmarks.

Reply:
Agree benchmark is needed.

Reply:
I tried saving 30000 graphs, with 30 nodes and ~90 edges, and feature size of 5 for edata and ndata. It just took about 3 seconds, which is totally acceptable.

Reply:
Ok, let's push this PR through first and see if it needs to be improved when we have more realistic benchmarks.